docs/distributions/ubuntu/upgrade.md

```diff
@@ -0,0 +1,63 @@
+# Upgrade
+
+Recommendations:
+
+* Perform upgrade from LTS to LTS.
+
+Warnings:
+
+* If you have important data and workspace, make sure that you have backup.
+* Make sure you have enough disk space.
+* **Ensure that you met all the below requirements so that you can recover in case of failed upgrade.**
+
+# Hardware Requirements
+
+In general, for troubleshooting you may need Internet access. If Wifi would work after a failure in upgrade process you **DO NOT** need the following requirements.
```

Comment:
This part though correct,now very confusing to read. You could firstly move this to the troubleshooting section instead of making this a seperate one. Secondly, you could make it simple by write something like :-

For troubleshooting you need internet access. It is possible that Wi-Fi may also not work for some users during troubleshooting. In that case, you shall also require <requirements of wired net>.